### PR TITLE
Feature/maze generation

### PR DESCRIPTION
**Pull request by swzwij**

### CODE REVIEW

File: Assets/Scripts/Maze/Generation/MazeGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+{
+    public static class MazeGenerator
+    {
+        private static Vector2 _mazeSize;
+
+        private static int _nextCellItteration;
+        private static int _completedCells;
+
+        private static readonly List<int> _seed = new();
+        private static readonly List<string> _fullSeed = new();
```
Comment (Minoord) [inline Assets/Scripts/Maze/Generation/MazeGenerator.cs:17]: I want to get the current seed!

File: Assets/Adaptive Performance.meta
Comment (BNS-MarkUlrich) [inline Assets/Adaptive Performance.meta:1]: Do we want to avoid spaces in folder names?

File: Assets/Scripts/Maze/Cells/MazeCell.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        private readonly Vector2[] _directions = { Vector2.right, Vector2.left, Vector2.up, Vector2.down };
+
+        private bool _isStart;
+        private bool _isEnd;
+
+        public CellState State { get; set; }
+
+        public Vector2 Position { get; set; }
+
+        public bool Start
```
Comment (BNS-MarkUlrich) [inline Assets/Scripts/Maze/Cells/MazeCell.cs]: I think it improves readability if you write bools as verbs. Aka `IsStart`, `CanStart`, etc

File: Assets/Scripts/Maze/Cells/MazeCell.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            {
+                _isStart = value;
+                SetWallsActive(false);
+                _floorRenderer.gameObject.SetActive(value);
+                _floorRenderer.material = _startMaterial;
+            }
+            get => _isStart; 
+        }
+
+        public bool End
```
Comment (BNS-MarkUlrich) [inline Assets/Scripts/Maze/Cells/MazeCell.cs]: I think it improves readability if you write bools as verbs. Aka `IsEnd`, `CanEnd`, etc

File: Assets/Scripts/Maze/Generation/LevelShowCase.cs
Comment (BNS-MarkUlrich) [inline Assets/Scripts/Maze/Generation/LevelShowCase.cs:1]: Can this be removed now? I don't mind if it stays (for the main menu, for example) but if it does, `rotate` should be renamed to a verb.

File: Assets/Scripts/Maze/Generation/MazeGeneratorDebug.cs
Comment (BNS-MarkUlrich) [inline Assets/Scripts/Maze/Generation/MazeGeneratorDebug.cs:1]: This can go

File: ProjectSettings/EditorBuildSettings.asset
Comment (BNS-MarkUlrich) [inline ProjectSettings/EditorBuildSettings.asset:1]: Please restore this to the version from develop.

File: ProjectSettings/Packages/com.unity.testtools.codecoverage/Settings.json
Comment (BNS-MarkUlrich) [inline ProjectSettings/Packages/com.unity.testtools.codecoverage/Settings.json:1]: Please restore this to the version from develop.

File: ProjectSettings/ProjectSettings.asset
Comment (BNS-MarkUlrich) [inline ProjectSettings/ProjectSettings.asset:1]: Please restore this to the version from develop.

File: ProjectSettings/SceneTemplateSettings.json
Comment (BNS-MarkUlrich) [inline ProjectSettings/SceneTemplateSettings.json:1]: Please restore this to the version from develop.

File: ProjectSettings/TimelineSettings.asset
Comment (BNS-MarkUlrich) [inline ProjectSettings/TimelineSettings.asset:1]: Please restore this to the version from develop.

**Review by BNS-MarkUlrich — CHANGES_REQUESTED** [review]

**Review by BNS-MarkUlrich — DISMISSED** [review]

**Pull request merged**